docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md

```diff
@@ -39,16 +39,68 @@ App Service Web Apps offers a fully managed platform optimized for hosting web a
+
+![End to End Docker DevOps Lifecycle Workflow](./media/image1-7.png)
+
+Developing with containers offers numerous advantages, especially when containers are used in production. The same container configuration is used to host the app in each environment in which it runs, from local dev machine to build and test systems to production. This greatly reduces the likelihood of defects resulting from differences in machine configuration or software versions. Developers can also use whatever tools they're most productive with, including operating system, since containers can run on any OS. In some cases, distributed applications involving many containers may be very resource-intensive to run on a single dev machine. In this scenario, it may make sense to upgrade to using Kubernetes and Azure Dev Spaces, covered in the next section.
+
+As portions of larger applications are broken up into their own smaller, independent *microservices*, additional design patterns can be used to improve app behavior. Instead of working directly with individual services, an *API gateway* can simplify access and decouple the client from its backend. Having separate service backends for different frontends also allows services to evolve in concert with their consumers. Common services can be accessed via a separate *sidecar* container, which might include common client connectivity libraries using the *ambassador* pattern.
+
+![Microservices sample architecture with several common design patterns noted.](./media/image1-10.png)
+
+[Learn more about design patterns to consider when building microservice-based systems.](https://docs.microsoft.com/azure/architecture/microservices/design/patterns)
+
+### Azure Kubernetes Service
+
+Azure Kubernetes Service (AKS) manages your hosted Kubernetes environment, making it quick and easy to deploy and manage containerized applications without container orchestration expertise. It also eliminates the burden of ongoing operations and maintenance by provisioning, upgrading, and scaling resources on demand, without taking your applications offline.
+
+AKS reduces the complexity and operational overhead of managing a Kubernetes cluster by offloading much of that responsibility to Azure. As a hosted Kubernetes service, Azure handles critical tasks like health monitoring and maintenance for you. Also, you pay only for the agent nodes within your clusters, not for the masters. As a managed Kubernetes service, AKS provides:
+
+- Automated Kubernetes version upgrades and patching.
+- Easy cluster scaling.
+- Self-healing hosted control plane (masters).
+- Cost savings - pay only for running agent pool nodes.
 
 With Azure handling the management of the nodes in your AKS cluster, you no longer need to perform many tasks manually, like cluster upgrades. Because Azure handles these critical maintenance tasks for you, AKS doesn't provide direct access (such as with SSH) to the cluster.
 
 #### Web App for Containers
```

Comment:
This section **Web App For Containers** and the following paragraph should be dropped.
> Web App for Containers from Azure App Service lets you bring your own Docker formatted container images and easily deploy and run them at scale with Azure. Web App for Containers supports both Linux and Windows containers and can scale to support large scale implementations. 

This is already covered in the section **App Service Web Apps for Containers**